crates/sui-framework/packages/sui-framework/sources/test/test_scenario.move

```diff
@@ -95,6 +109,101 @@ public struct TransactionEffects has drop {
     num_user_events: u64,
 }
 
+//
+// `TxContextBuilder` api
+//
+
+/// Create a new `TxContextBuilder` with the given `sender` address.
+/// Also provides default for all other fields.
+public fun new_tx_context_builder(sender: address): TxContextBuilder {
+    TxContextBuilder {
+        sender,
+        epoch: 0,
+        epoch_timestamp_ms: 0,
+        ids_created: 0,
+        rgp: option::some(700),
+        gas_price: 1_000,
+        gas_budget: 100_000_000,
+        sponsor: option::none(),
+    }
+}
+
+/// Create a `TxContextBuilder` from an existing `TxContext`.
+public fun from_context(ctx: &TxContext): TxContextBuilder {
```

Comment:
What about adding a `tx_builder_from_scenario` function as well? It may allow for some more ergonomic programming patterns e.g., instead of

```move
let ctx_builder = test_scenario::from_context(scenario.ctx())
  .set_gas_price(..)
  .set_gas_budget(..);
```
I can instead do

```move
let ctx_builder = scenario
  .tx_builder_from_scenario()
  .set_gas_price(..)
  .set_gas_budget(..);
```

Reply:
like it! will do